### PR TITLE
Make July 2023 news public

### PR DESCRIPTION
**Pull request by cticenhour**
Also adds August news template.

@idaholab/moose-ccb -- Please add any last minute additions from the last month. This edition is a bit light, so I imagine there are several!

Refs #11447

### CODE REVIEW

**Comment by YaqiWang** [comment]
Just notice this. I will add one entry about the console time format that @GiudGiud suggested me.

File: modules/doc/content/newsletter/2023/2023_07.md [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
 
 ## libMesh-level Changes
 
+### `2023.07.26` Update
+
+- `HIERARCHIC` and `L2_HIERARCHIC` Finite Element families now support
+  prism elements
+- Added `NumericVector::l2_norm_diff()`, to enable calculation of
+  vector differences without creating a temporary vector.
+- Fixes for `MeshedHole` (and thereby [XYDelaunayMeshGenerator.md]
```
Comment (cticenhour) [inline modules/doc/content/newsletter/2023/2023_07.md:13]:
```suggestion
- Fixes for `MeshedHole` (and thereby [XYDelaunayGenerator.md]
```

File: modules/doc/content/newsletter/2023/2023_07.md [hunk 2]
```diff
@@ -1,14 +1,81 @@
...
+precisely; (2) model inadequacy due to missing physics compared to the experiments; and
+(3) noise in the observed experimental data itself. To quantify these sources of uncertainties,
+Bayesian UQ capabilities have been implemented in the [MOOSE Stochastic Tools Module (STM)](modules/stochastic_tools/index.md).
+When dealing with computationally expensive finite element models, it is known that Bayesian UQ can become
+expensive. To this end, the Bayesian UQ capabilties in the STM are massively parallelizable, reducing
+the overall elaspsed time for the UQ with the aid of parallel computing. Some salient features of these
+new UQ implementation include:
+
+- Parallel Markov chain Monte Carlo (MCMC) methods under the parallel Metropolis-Hastings and Ensemble class of samplers.
+- Parallel scalability tested upto 50 parallel proposals.
```
Comment (lindsayad) [inline modules/doc/content/newsletter/2023/2023_07.md:56]:
```suggestion
- Parallel scalability tested up to 50 parallel proposals.
```

**Review by lindsayad — CHANGES_REQUESTED** [review]

File: modules/doc/content/newsletter/2023/2023_07.md [hunk 3]
```diff
@@ -1,14 +1,81 @@
...
+MOOSE now supports p-refinement, e.g. refining or coarsening the order of
+the finite element basis functions. In order to use p-refinement, you
+can simply set `switch_h_to_p_refinement=true` in the `Adaptivity` block.
+As suggested by the parameter name, only one of h- or p-refinement may be used
+in a single simulation.
+
+### Console time formatting
+
+MOOSE now has a parameter for [Console](outputs/Console.md) outputs, [!param](/Outputs/Console/time_format), to control
+how times are printed during transient simulations. Internally, MOOSE uses seconds as its unit of
```
[the remark this answers is not in the document]
Comment (cticenhour) [reply modules/doc/content/newsletter/2023/2023_07.md:16]: Yeah good point - this is only a convenience addition to the printout. How about "Internally, MOOSE has no default unit for time, but, with this change, MOOSE can now...."?

[the remark this answers is not in the document]
Comment (cticenhour) [reply modules/doc/content/newsletter/2023/2023_07.md]: Resolved in [1c31713](https://github.com/idaholab/moose/pull/25062/commits/1c31713c928ae912e5a19f18a27ecb8693352a2e)

File: modules/doc/content/newsletter/2023/2023_07.md [hunk 4]
```diff
@@ -1,14 +1,81 @@
...
+  vector differences without creating a temporary vector.
+- Fixes for `MeshedHole` (and thereby [XYDelaunayGenerator.md]
+  calculations and error reporting when an input mesh contains both 2D
+  and 1D elements.
+- Metis and Parmetis are now IsoGeometric-Analysis-aware, and produce
+  much better-scaling partitionings of IGA meshes.
+- More understandable error reporting if a user's environment mixes
+  `DEBUG` and `NDEBUG` flags into the same build.
+
 ## PETSc-level Changes
```
Comment (cticenhour) [inline modules/doc/content/newsletter/2023/2023_07.md:80]:
```suggestion
```

Should remove since PETSc update is happening in August news.

**Review by lindsayad — APPROVED** [review]

**Pull request merged**